src/rules/no-unused-modules.js

```diff
@@ -207,38 +207,39 @@ const fileIsInPkg = file => {
   const { path, pkg } = readPkgUp.sync({cwd: file, normalize: false})
   const basePath = dirname(path)
 
-  if (pkg.private === true) {
-    return false
-  }
+  const checkPkgField = pkgField => {
+    if (typeof pkgField === 'string') {
+      if (join(basePath, pkgField) === file) {
+        return true
+      }
+    }
 
-  if (pkg.bin && typeof pkg.bin === 'string') {
-    if (join(basePath, pkg.bin) === file) {
-      return true
+    if (typeof pkgField === 'object') {
+      const pkgFieldFiles = Object.values(pkgField).map(value => join(basePath, value))
+      if (pkgFieldFiles.includes(file)) {
+        return true
+      }
     }
   }
 
-  if (pkg.bin && typeof pkg.bin === 'object') {
-    const binFiles = Object.values(pkg.bin).map(value => join(basePath, value))
-    if (binFiles.includes(file)) {
-      return true
-    }
+  if (pkg.private === true) {
+    return false
   }
 
-  if (pkg.browser && typeof pkg.browser === 'string') {
-    if (join(basePath, pkg.browser) === file) {
+  if (pkg.bin) {
+    if (checkPkgField(pkg.bin)) {
       return true
     }
   }
 
-  if (pkg.browser && typeof pkg.browser === 'object') {
-    const browserFiles = Object.values(pkg.browser).map(value => join(basePath, value))
-    if (browserFiles.includes(file)) {
+  if (pkg.browser) {
+    if (checkPkgField(pkg.browser)) {
       return true
     }
   }
 
   if (pkg.main) {
-    if (join(basePath, pkg.main) === file) {
+    if (checkPkgField(pkg.main)) {
```

Comment:
the only remaining problem here is that main can’t be an object :-/

Reply:
Maybe I don't see the issue here. `main` will either return `true` within the first `if` branch or just return at the end of the function.
Would it be better to return `false` in the first `if` branch to make `main` not even pass the `if` branch for object handling?

Reply:
Yes, that's what I'm suggesting. an object "main" field is just invalid and shouldn't be considered.